gnucash/import-export/import-main-matcher.cpp

```diff
@@ -192,32 +195,95 @@ defer_bal_computation (GNCImportMainMatcher *info, Account* acc)
     }
 }
 
+
+// Custom iterator class
+class GtkTreeModelIterator
+{
+public:
+    using iterator_category = std::forward_iterator_tag;
+    using difference_type = ptrdiff_t;
+    GtkTreeModelIterator(GtkTreeModel* model, std::optional<GtkTreeIter> iter)
+        : m_model(model), m_iter(iter) {}
+    GtkTreeModelIterator& operator++()
+    {
+        if (m_iter.has_value() && !gtk_tree_model_iter_next(m_model, &*m_iter))
+            m_iter = std::nullopt;
+        return *this;
+    }
+    template <typename T>
+    T get (int column) {
+        g_return_val_if_fail (m_iter.has_value(), T());
+        gpointer rv;
+        gtk_tree_model_get(m_model, &*m_iter, column, &rv, -1);
```

Comment:
I'm using `std::nullopt` to denote the `end()` iterator. The original suggestions use `GtkTreeIter()` as the constructor for the `end()` iterator but I figure nullopt is safer, especially needing to test equality. This means we need the weird `&*m_iter` construct.

Reply:
You could use `&m_iter.value()` instead.